src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -268,7 +291,18 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                         unsupportedSortingParameters = searchOptions.UnsupportedSortingParams;
                     }
 
-                    return new SearchResult(resources, searchOptions.UnsupportedSearchParams, unsupportedSortingParameters, moreResults ? newContinuationId.Value.ToString(CultureInfo.InvariantCulture) : null, _isResultPartial);
+                    // Continuation token prep
+                    var continuationToken = new ContinuationToken(new object[]
```

Comment:
We should not allocate this if `!moreResults`. 
Also, if you should allocate it you pass in either `new[] { newContinuationId }` or `new [] { sortExpr, newContinuationId }`.